lib/spack/spack/user_environment.py

```diff
@@ -102,6 +102,9 @@ def environment_modifications_for_specs(
 
     # Static environment changes (prefix inspections)
     for s in reversed(topo_ordered):
+        for p in getattr(s.package, "cmake_prefix_paths", []):
+            env.prepend_path("CMAKE_PREFIX_PATH", p)
+
         static = environment.inspect_path(
             s.prefix, prefix_inspections(s.platform), exclude=environment.is_system_path
```

Comment:
Or does it maybe make sense to modify prefix_inspections to use get_cmake_prefix_path ?

https://github.com/spack/spack/blob/e9e6eb613b752bb3e40f324000896b4c2668587f/lib/spack/spack/user_environment.py#L45

Reply:
Maybe both of my comments are what you were referring to in https://github.com/spack/spack/issues/38265#issuecomment-1932684337 but I only had a brief look at this